Fix and extend color documentation (#1773)
nushell/nushell.github.io: book/coloring_and_theming.md

- `none`

> - `other`

Reviewer: `other` isn't valid. This is the list.
```nushell
❯ table --list                                                                                                              27  01:38:06 PM
╭────┬────────────────╮
│  0 │ basic          │
│  1 │ compact        │
│  2 │ compact_double │
│  3 │ default        │
│  4 │ heavy          │
│  5 │ light          │
│  6 │ none           │
│  7 │ reinforced     │
│  8 │ rounded        │
│  9 │ thin           │
│ 10 │ with_love      │
│ 11 │ psql           │
│ 12 │ markdown       │
│ 13 │ dots           │
│ 14 │ restructured   │
│ 15 │ ascii_rounded  │
│ 16 │ basic_compact  │
╰────┴────────────────╯
```

Author: I did not touch these. Can we handle the value updates separately?

I would prefer not to expand the scope of this format, text, and desc improvements and expansion with value doc updates/fixes.

Reviewer: ok, i'm ok with incremental changes. it just needs to be done at some point because both lists are incomplete.

Author: I created https://github.com/nushell/nushell.github.io/pull/1774

## Color Configuration

The color configuration is defined in `$env.config.color_config`. The current configuration can be printed with:

```nu

> $env.config.color_config | sort

Reviewer: sort is not required

Author: It's not required, but useful. Without it the output is not sorted.

```

The color and style-attributes can be declared in multiple alternative formats.

- `r` - normal color red's abbreviation
- `rb` - normal color red's abbreviation with bold attribute
- `red` - normal color red
- `red_bold` - normal color red with bold attribute
- `"#ff0000"` - "#hex" format foreground color red (quotes are required)
- `{ fg: "#ff0000" bg: "#0000ff" attr: b }` - "full #hex" format foreground red in "#hex" format with a background of blue in "#hex" format with an attribute of bold abbreviated.

> - `{|x| 'yellow' }` - closure returning a valid string

Reviewer: it's really not just any string. it has to be a color listed in `ansi --list`. same for red_bold above.

Author: ```suggestion
- `{|x| 'yellow' }` - closure returning a string with one of the color representations listed above
```

- `{ fg: "#ff0000" bg: "#0000ff" attr: b }` - "full #hex" format foreground red in "#hex" format with a background of blue in "#hex" format with an attribute of bold abbreviated.
- `{|x| 'yellow' }` - closure returning a string with one of the color representations listed above
- `{|x| { fg: "#ff0000" bg: "#0000ff" attr: b } }` - closure returning a valid record

### Attributes

---

> 

Reviewer: The list of color "code" and "name" is incomplete. You can see the complete list with `ansi --list | first 133`

Author: https://github.com/nushell/nushell.github.io/pull/1775

- background of blue in "#hex" format
- attribute of bold abbreviated

### Closure

> Note that closures are only executed for table output. They do not work for `shape_` configurations, nor when output as the value type itself, or as a list of such value.

Author: ```suggestion
Note: Closures are only executed for table output. They do not work in other contexts like for `shape_` configurations, when printing a value directly, or as a value in a list.
```

For example:

```nu
$env.config.color_config.filesize = {|x| if $x == 0b { 'dark_gray' } else if $x < 1mb { 'cyan' } else { 'blue' } }

> $env.config.color_config.filesize = {|x| if $x == 0b { 'dark_gray' } else if $x < 1mb { 'cyan' } else { 'blue_bold' } }

Reviewer: i'm not sure why there are two rows here each setting filesize but differently?

Author: oh I meant to include a bool format closure

Author: ```suggestion
$env.config.color_config.bool = {|x| if $x { 'green' } else { 'light_red' } }
```

{a:true,b:false,c:0mb,d:0.5mb,e:10mib}
```

prints

```nu
╭───┬───────────╮
│ a │ true      │
│ b │ false     │
│ c │ 0 B       │
│ d │ 488.3 KiB │
│ e │ 10.0 MiB  │
╰───┴───────────╯
```

> with a green `true`, a red `false`, a grey `0 B`, a cyan `488.3 KiB`, and a blue `10.0 MiB`.

Reviewer: The color names don't match the definition above
```suggestion
with a green `true`, a light_red `false`, a dark_grey `0 B`, a cyan `488.3 KiB`, and a blue `10.0 MiB`.
```

Author: I deliberately did not use the symbol names. As a description for humans, I chose the simplest description of color names. Because there are no different heues displayed in this output. Just one kind of red and one kind of grey.

If you want it specific, I would at least use normal speak and not the symbols

```suggestion
with a green `true`, a light red `false`, a dark grey `0 B`, a cyan `488.3 KiB`, and a blue `10.0 MiB`.
```

Reviewer: I just thought it was confusing because you're saying it's a different color than you specified above. I could live with your suggestion because at least someone can compare and agree that the setting worked based on the names alone.
